src/java.base/windows/classes/sun/util/locale/provider/HostLocaleProviderAdapterImpl.java

```diff
@@ -366,7 +366,14 @@ public int getFirstDayOfWeek(Locale locale) {
 
             @Override
             public int getMinimalDaysInFirstWeek(Locale locale) {
-                return 0;
+                int firstWeek = getCalendarDataValue(
+                        removeExtensions(locale).toLanguageTag(),
+                        CD_FIRSTWEEKOFYEAR);
+                return switch (firstWeek) {
+                    case 1 -> 7;
```

Comment:
Would it be good to use constants or enum instead of literal, or maybe at least a note for the case numbers.

Reply:
Could be better by making them as enums, but there are other locations similarly using ints. I added extra comments to explain those ints for better readability.